src/library/scala/annotation/InternalApi.scala

```diff
@@ -0,0 +1,28 @@
+/*
+ * Scala (https://www.scala-lang.org)
+ *
+ * Copyright EPFL and Lightbend, Inc.
+ *
+ * Licensed under Apache License 2.0
+ * (http://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package scala.annotation
+
+/**
+ * Marks APIs that are considered internal to the library and may change at any point in time without any warning.
+ *
+ * For example, this annotation should be used when the Scala `private[foo]` access restriction is used,
```

Comment:
I'm not 100% sure this annotation adds a lot of value. From the java (tooling) perspective, it doesn't change anything: the bytecode is public, and the annotation is not visible (it's only stored in the Scala signature). For Scala tooling, if a Scala tool can read this annotation it can instead read the `private[q]` qualifier. Asking that "everyone using `private[q]` should add the `@InternalApi` annotation" sounds like a limitation on the tooling front.

Reply:
> the annotation is not visible (it's only stored in the Scala signature)

Ah, then I agree this doesn't work.

Is it possible to define an annotation that _is_ visible in the bytecode, in Scala?

Alternatively, any other way to define a Java annotation in the standard library?  Is defining it in Java acceptable?